### PR TITLE
Add support for user headers

### PR DESCRIPTION
**Pull request by michaelbausor**
User headers are required for Spanner

cc @dwsupplee @jdpedrie

### CODE REVIEW

File: src/ApiCallable.php [hunk 1]
```diff
@@ -168,16 +168,23 @@ private static function createGrpcStreamingApiCall($callable, $grpcStreamingDesc
...
                 throw new InvalidArgumentException('Metadata argument is not found.');
             } else {
                 $metadata = $params[self::GRPC_CALLABLE_METADATA_INDEX];
-                $headers = $headerDescriptor->getHeader();
+                $headers = [];
+                if (!is_null($userHeaders)) {
+                    $headers = $userHeaders;
+                }
+                if (!is_null($headerDescriptor)) {
+                    $headers = array_merge($headers, $headerDescriptor->getHeader());
```
Comment (garrettjonesgoogle) [inline src/ApiCallable.php]: Maybe add a note somewhere that user headers can't override x-goog-api-client

File: tests/ApiCallableTest.php [hunk 2]
```diff
@@ -496,6 +496,39 @@ public function testCustomHeader()
...
+            $callSettings,
+            ['headerDescriptor' => $headerDescriptor]
+        );
+        $resources = $apiCall(new MockPageStreamingRequest(), [], []);
+        $actualCalls = $stub->getReceivedCalls();
+        $this->assertEquals(1, count($actualCalls));
+        $expectedMetadata = [
+            'x-goog-api-client' => ['gl-php/5.5.0 gccl/0.0.0 gapic/0.9.0 gax/1.0.0 grpc/1.0.1'],
+            'google-cloud-resource-prefix' => ['my-database'],
+        ];
```
Comment (garrettjonesgoogle) [inline tests/ApiCallableTest.php]: Maybe also add a test that makes sure the user can't override x-goog-api-client

File: src/ApiCallable.php (hunk 1 above)
Comment (michaelbausor) [inline src/ApiCallable.php]: Done

File: tests/ApiCallableTest.php (hunk 2 above)
Comment (michaelbausor) [inline tests/ApiCallableTest.php]: Done

**Comment by michaelbausor** [comment]
PTAL

**Comment by garrettjonesgoogle** [comment]
LGTM

**Pull request merged**